File: app/src/main/java/mega/privacy/android/app/arch/BaseRxViewModel.java
package mega.privacy.android.app.arch;

import androidx.lifecycle.AndroidViewModel;
import io.reactivex.rxjava3.disposables.CompositeDisposable;
import io.reactivex.rxjava3.disposables.Disposable;
import mega.privacy.android.app.MegaApplication;

public abstract class BaseRxViewModel extends AndroidViewModel {
    protected CompositeDisposable composite = new CompositeDisposable();

    public BaseRxViewModel() {
        super(MegaApplication.getInstance());
    }

    protected void add(Disposable disposable) {
        composite.add(disposable);
    }

    @Override protected void onCleared() {
        super.onCleared();

        composite.clear();
    }
}

File: app/src/main/java/mega/privacy/android/app/arch/BaseRxViewModel.kt
package mega.privacy.android.app.arch

import androidx.lifecycle.ViewModel
import io.reactivex.rxjava3.disposables.CompositeDisposable
import io.reactivex.rxjava3.disposables.Disposable

abstract class BaseRxViewModel : ViewModel() {

    protected var composite = CompositeDisposable()

    protected fun add(disposable: Disposable?) {
        composite.add(disposable)
    }

    override fun onCleared() {
        super.onCleared()
        composite.clear()
    }
}
